microservices/google-api/google_mybusiness.py

```diff
@@ -0,0 +1,435 @@
+            logger.info("first time loading {}: {}"
+                        .format(account['name'], loc['name']))
+
+        # If it is loaded with some data for this ID, use that date plus
+        # one day as the start_date.
+        if (last_loaded_date is not None
+                and last_loaded_date.isoformat() >= start_date):
+            start_date = last_loaded_date + timedelta(days=1)
+            start_date = start_date.isoformat()
+
+        start_time = start_date + 'T00:00:00Z'
+
+        # if an end_date is defined in the config file, use that date
+        date_api_upper_limit = (
+            datetime.datetime.today().date() - timedelta(days=2)).isoformat()
+        end_date = account['end_date']
+        if end_date == '':
+            # The most recent data available is from two days ago
+            end_date = date_api_upper_limit
+        if end_date > date_api_upper_limit:
+            logger.warning("The end_date is more recent than 2 days ago.")
+
+        end_time = end_date + 'T00:00:00Z'
+
+        # if start and end times are same, then there's no new data
+        if start_time == end_time:
+            logger.info("Redshift already contains the latest avaialble data.")
+            continue
+
+        logger.debug("Querying range from {0} to {1}"
+                     .format(start_date, end_date))
+
+        # the API call must construct each metric request in a list of dicts
+        metricRequests = []
+        for metric in config_metrics:
+            metricRequests.append(
+                {
+                    'metric': metric,
+                    'options': 'AGGREGATED_DAILY'
+                    }
+                )
+
+        bodyvar = {
+            'locationNames': ['{0}'.format(location_uri)],
+            'basicRequest': {
+                # https://developers.google.com/my-business/reference/rest/v4/Metric
+                'metricRequests': metricRequests,
+                # https://developers.google.com/my-business/reference/rest/v4/BasicMetricsRequest
+                # The maximum range is 18 months from the request date.
+                'timeRange': {
+                    'startTime': '{0}'.format(start_time),
+                    'endTime': '{0}'.format(end_time)
+                    }
+                }
+            }
+
+        logger.debug("Request body:\n{0}"
+                     .format(json.dumps(bodyvar, indent=2)))
+
+        # retrieves the request for this location.
+        try:
+            reportInsights = \
+                service.accounts().locations().\
+                reportInsights(body=bodyvar, name=account_uri).execute()
+        except googleapiclient.errors.HttpError as e:
+            logger.info("Request contains an invalid argument. Skipping.")
+            continue
+
+        # # dump reportInsights to a file for manual review
+        # with open(location_uri + 'json', 'w', encoding='utf-8') as f:
+        #     json.dump(reportInsights, f, indent=2)
```

Comment:
If this is commented out, can it just be removed?

Reply:
Cleaned up in the last commit.